glusterd2/commands/volumes/volume-option.go

```diff
@@ -113,13 +134,25 @@ func xlatorAction(c transaction.TxnCtx, txnOp txnOpType, volOp volumeOpType) err
 			reqOptions[key] = op.DefaultValue
 		}
 
-	}
-	var volinfo volume.Volinfo
-	if err := c.Get("volinfo", &volinfo); err != nil {
-		return err
+	case xlator.VolumeStart:
+		fallthrough
+	case xlator.VolumeStop:
+		//start the daemon, if daemon option default value is "on",
+		//or if particular daemon option is set to "on" in volume option
+		//stop daemon, if there are no started volumes to manage
+		for _, actor := range xlator.GetOptActors() {
+			if txnOp == txnDo {
+				fn = actor.Do
+			} else {
+				fn = actor.Undo
+			}
+			if err := fn(&volinfo, "", "", volOp, c.Logger()); err != nil {
+				return err
+			}
+		}
```

Comment:
Shouldn't you return here ? 

```go
return fn(&volinfo, "", "", volOp, c.Logger()
```
The `fn()` is called again below. And that is only for `VolumeSet` and `VolumeReset` cases